include/boost/math/tools/polynomial_gcd.hpp

```diff
@@ -14,8 +14,57 @@
 #include <algorithm>
 #include <type_traits>
 #include <boost/math/tools/polynomial.hpp>
+
+#ifndef BOOST_MATH_STANDALONE
 #include <boost/integer/common_factor_rt.hpp>
 
+// std::gcd was introduced in C++17
+#elif (__cplusplus > 201700L || _MSVC_LANG > 201700L)
+
+#define BOOST_MATH_CXX17_NUMERIC
+#include <numeric>
+#include <utility>
+#include <iterator>
+#include <type_traits>
+#include <boost/math/tools/assert.hpp>
+
+namespace boost::integer {
+
+template <typename Iter, typename T = typename std::iterator_traits<I>::value_type>
+std::pair<T, Iter> gcd_range(Iter first, Iter last) noexcept(std::is_arithmetic_v<T>)
+{
+    using std::gcd;
+    BOOST_MATH_ASSERT(first != last);
+
+    T d = *first;
+    ++first;
+    while (d != T(1) && first != last)
+    {
+        d = gcd(d, *first);
+        ++first;
+    }
+    return std::make_pair(d, first);
+}
+
+namespace gcd_detail {
+
+template <typename EuclideanDomain>
+inline EuclideanDomain Euclid_gcd(EuclideanDomain a, EuclideanDomain b) noexcept(std::is_arithmetic_v<EuclideanDomain>)
+{
+    using std::swap;
+    while (b != EuclideanDomain(0))
+    {
+        a %= b;
+        swap(a, b);
+    }
+    return a;
+}
+
+} // namespace gcd_detail
+} // namespace boost::integer
+#error polynomial gcd can only be used in standalone mode with C++17 or higher
```

Comment:
This #error appears to be misplaced - missing a #else branch?

More pertinently, this header is included by polynomial.hpp, so we don't want hard #error's here, probably better to static_assert in it's interface somewhere if the gcd code is actually instantiated.  BTW I'm *reasonably* sure that static_assert's need to depend on the template parameter, if you do a `static_assert(false, "message")` then the compiler get get ultra-cute and realises that the assert will always trigger and issue an error even if the template is never instantiated.  That's why I've used `static_assert(sizeof(T) == 0, "message")` in the past, even though it's much less readable.